File: server/Models/Group.cs
using System.ComponentModel.DataAnnotations;
using System.Runtime.InteropServices.JavaScript;

namespace alg_dashboard_server.Models;

public enum GroupType
{
    Group,
    Individual,
    Intensive
}
public class Group
{
    public int Id { get; init; }
    [MaxLength(100)] public required string Name { get; set; }
    public required DateOnly StartDate { get; set; }
    public required DayOfWeek LessonDay { get; set; }
    public required TimeOnly LessonTime { get; set; }
    public required string BackOfficeUrl { get; set; }
    public required int CourseId { get; set; }
    public required int LessonsAmount { get; set; }
    public required GroupType Type { get; set; }
    public Course Course { get; init; } = null!;
    
    public required int UserId { get; set; }
    public User User { get; init; } = null!;
    
    public List<GroupStudent> GroupStudents { get; init; } = [];
    public List<Lesson> Lessons { get; set; } = [];
}

File: client/src/types/group.ts
import { ILesson } from './lesson'
import { IStudent } from './student'
import { IUser } from './user'

export enum DayOfWeek {
  'Воскресенье' = 0,
  'Понедельник' = 1,
  'Вторник' = 2,
  'Среда' = 3,
  'Четверг' = 4,
  'Пятница' = 5,
  'Суббота' = 6,
}

export enum GroupColors {
  'bg-blue-500' = 0,
  'bg-green-500' = 1,
  'bg-purple-500' = 2,
  'bg-orange-500' = 3,
}

export interface IGroup {
  id: number
  name: string
  course: string
  teacher: IUser
  startDate: string
  lessonDay: DayOfWeek
  lessonTime: string
  backOfficeUrl: string
  students: IStudent[]
  lessons: ILesson[]
}
